core/services/gateway/network/wsserver.go

```diff
@@ -62,9 +66,23 @@ func NewWebSocketServer(config *WebSocketServerConfig, acceptor ConnectionAccept
 	return server
 }
 
+func (s *webSocketServer) GetPort() int {
+	return s.listener.Addr().(*net.TCPAddr).Port
+}
+
 func (s *webSocketServer) handleRequest(w http.ResponseWriter, r *http.Request) {
 	authHeader := r.Header.Get(WsServerHandshakeAuthHeaderName)
-	authBytes, _ := base64.StdEncoding.DecodeString(authHeader)
+	if len(authHeader) > HandshakeEncodedAuthHeaderMaxLen {
+		s.lggr.Errorw("received auth header is too large", "len", len(authHeader))
+		w.WriteHeader(http.StatusUnauthorized)
```

Comment:
To me it seems `http.BadRequest` is the better code here.

Reply:
I would keep `StatusUnauthorized` for when the request was proper, but the authorization has failed.